### PR TITLE
feat: added procurement and agreement tests

### PR DESCRIPTION
**Pull request by CesarPerez**
### JIRA link
https://crowncommercialservice.atlassian.net/browse/SCAT-2518
https://crowncommercialservice.atlassian.net/browse/SCAT-2520

### Change description
Tests for procurement and agreement controllers

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Check and send page updated
- [ ] UI changes look good on mobile

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by prathap-cognizant — APPROVED** [review]

File: src/test/routes/features/agreement/controllers/agreement.test.ts [hunk 1]
```diff
@@ -1,13 +1,86 @@
...
-  describe('on GET', () => {
-    it('should render `Choose a commercial agreement` page when everything is fine', async () => {
-      await request(app)
-        .get('/projects/choose-agreement')
-        .expect(res => expect(res.status).to.equal(302)) // it's 302 bcz we need to mock the conclave data, then we will get 200. Now it's going to login screen.
-    })
-  })
+  let parentApp;
+  const dummyJWT = 'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2Mzc3NTU1NTUsImp0aSI6IjQ0YWVjN2Q0LTAxY2UtNGEwNS05MWE3LTU3ZWExNmUxZjFjNiIsInVpZCI6IjEwMzgiLCJjYWxsZXIiOiJ1c2VyIiwiY2lpT3JnSWQiOiI1NTY4OTE2MzM2MTk4NTE2MDMiLCJyb2xlcyI6WyJDQVRfQURNSU5JU1RSQVRPUiIsIkNBVF9VU0VSIl0sInN1YiI6ImNlc2FyLmZlcm5hbmRlekBjb2duaXphbnQuY29tIiwiZXhwIjoxNjM3NzU2NDU1LCJpc3MiOiJDQ1MiLCJhdWQiOiJ6eWljckRhMG9Kc0g0aFVMSVdUTmRhZHhRVjQ3N3c0NSJ9.eveTQA_33H2WOWig1eK0O1n8kOODQCUrXVQWxGaI-zVKw4Jjh9FPX9dzireNGzLxfkJYlb1IOW-4LR2g6LldUIs2RY9-2-rBo9ZIGTgcRjgR_uvM-UnhMTfbuTJWfDK8r7XQ1A4qNOywlAQ8xDjhNzrHwA8ZreFJVE6ijiYJHdMyNgIwJKtvNLE9k1TksPf0eDw90QhYMI3uJaAg4wRDgNVV8gyr3Gacsac6phCeKlfMJRlzrS_k4OcKG9nqQgfpHhqvvO-58TUY72rPAWJQ7m4xSiFXD2B8VR-n7ZgleRyvQb3K47Eb-4Tiu70kcPxtEcczxhQTfZB8BpUg3-y_hA'
```
Comment (daimdev) [inline src/test/routes/features/agreement/controllers/agreement.test.ts]: This consist of data

**Review by ulysses-cognizant — APPROVED** [review]

File: src/test/routes/features/agreement/controllers/lot.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+import { expect } from 'chai';
+import request from 'supertest';
+import { app } from '../../../../../main/app';
+import nock from 'nock'
+import express from 'express';
+
+describe('Choose a lot agreement page', () => {
+  let parentApp;
+  const dummyJWT = 'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2Mzc3NTU1NTUsImp0aSI6IjQ0YWVjN2Q0LTAxY2UtNGEwNS05MWE3LTU3ZWExNmUxZjFjNiIsInVpZCI6IjEwMzgiLCJjYWxsZXIiOiJ1c2VyIiwiY2lpT3JnSWQiOiI1NTY4OTE2MzM2MTk4NTE2MDMiLCJyb2xlcyI6WyJDQVRfQURNSU5JU1RSQVRPUiIsIkNBVF9VU0VSIl0sInN1YiI6ImNlc2FyLmZlcm5hbmRlekBjb2duaXphbnQuY29tIiwiZXhwIjoxNjM3NzU2NDU1LCJpc3MiOiJDQ1MiLCJhdWQiOiJ6eWljckRhMG9Kc0g0aFVMSVdUTmRhZHhRVjQ3N3c0NSJ9.eveTQA_33H2WOWig1eK0O1n8kOODQCUrXVQWxGaI-zVKw4Jjh9FPX9dzireNGzLxfkJYlb1IOW-4LR2g6LldUIs2RY9-2-rBo9ZIGTgcRjgR_uvM-UnhMTfbuTJWfDK8r7XQ1A4qNOywlAQ8xDjhNzrHwA8ZreFJVE6ijiYJHdMyNgIwJKtvNLE9k1TksPf0eDw90QhYMI3uJaAg4wRDgNVV8gyr3Gacsac6phCeKlfMJRlzrS_k4OcKG9nqQgfpHhqvvO-58TUY72rPAWJQ7m4xSiFXD2B8VR-n7ZgleRyvQb3K47Eb-4Tiu70kcPxtEcczxhQTfZB8BpUg3-y_hA'
```
Comment (daimdev) [inline src/test/routes/features/agreement/controllers/lot.test.ts]: This consit of personal data which is being exposed

File: src/test/routes/features/agreement/controllers/selectedAgreement.test.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+import { expect } from 'chai';
+import request from 'supertest';
+import { app } from '../../../../../main/app';
+import nock from 'nock'
+import express from 'express';
+
+describe('Choose a lot agreement page', () => {
+  let parentApp;
+  const dummyJWT = 'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2Mzc3NTU1NTUsImp0aSI6IjQ0YWVjN2Q0LTAxY2UtNGEwNS05MWE3LTU3ZWExNmUxZjFjNiIsInVpZCI6IjEwMzgiLCJjYWxsZXIiOiJ1c2VyIiwiY2lpT3JnSWQiOiI1NTY4OTE2MzM2MTk4NTE2MDMiLCJyb2xlcyI6WyJDQVRfQURNSU5JU1RSQVRPUiIsIkNBVF9VU0VSIl0sInN1YiI6ImNlc2FyLmZlcm5hbmRlekBjb2duaXphbnQuY29tIiwiZXhwIjoxNjM3NzU2NDU1LCJpc3MiOiJDQ1MiLCJhdWQiOiJ6eWljckRhMG9Kc0g0aFVMSVdUTmRhZHhRVjQ3N3c0NSJ9.eveTQA_33H2WOWig1eK0O1n8kOODQCUrXVQWxGaI-zVKw4Jjh9FPX9dzireNGzLxfkJYlb1IOW-4LR2g6LldUIs2RY9-2-rBo9ZIGTgcRjgR_uvM-UnhMTfbuTJWfDK8r7XQ1A4qNOywlAQ8xDjhNzrHwA8ZreFJVE6ijiYJHdMyNgIwJKtvNLE9k1TksPf0eDw90QhYMI3uJaAg4wRDgNVV8gyr3Gacsac6phCeKlfMJRlzrS_k4OcKG9nqQgfpHhqvvO-58TUY72rPAWJQ7m4xSiFXD2B8VR-n7ZgleRyvQb3K47Eb-4Tiu70kcPxtEcczxhQTfZB8BpUg3-y_hA'
```
Comment (daimdev) [inline src/test/routes/features/agreement/controllers/selectedAgreement.test.ts]: This consist of personal data which is being exposed

File: src/test/routes/features/procurement/controllers/procurement.test.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+import { expect } from 'chai';
+import request from 'supertest';
+import { app } from '../../../../../main/app';
+import nock from 'nock'
+import express from 'express';
+
+describe('Procurement page', () => {
+    let parentApp;
+    const dummyJWT = 'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2Mzc3NTU1NTUsImp0aSI6IjQ0YWVjN2Q0LTAxY2UtNGEwNS05MWE3LTU3ZWExNmUxZjFjNiIsInVpZCI6IjEwMzgiLCJjYWxsZXIiOiJ1c2VyIiwiY2lpT3JnSWQiOiI1NTY4OTE2MzM2MTk4NTE2MDMiLCJyb2xlcyI6WyJDQVRfQURNSU5JU1RSQVRPUiIsIkNBVF9VU0VSIl0sInN1YiI6ImNlc2FyLmZlcm5hbmRlekBjb2duaXphbnQuY29tIiwiZXhwIjoxNjM3NzU2NDU1LCJpc3MiOiJDQ1MiLCJhdWQiOiJ6eWljckRhMG9Kc0g0aFVMSVdUTmRhZHhRVjQ3N3c0NSJ9.eveTQA_33H2WOWig1eK0O1n8kOODQCUrXVQWxGaI-zVKw4Jjh9FPX9dzireNGzLxfkJYlb1IOW-4LR2g6LldUIs2RY9-2-rBo9ZIGTgcRjgR_uvM-UnhMTfbuTJWfDK8r7XQ1A4qNOywlAQ8xDjhNzrHwA8ZreFJVE6ijiYJHdMyNgIwJKtvNLE9k1TksPf0eDw90QhYMI3uJaAg4wRDgNVV8gyr3Gacsac6phCeKlfMJRlzrS_k4OcKG9nqQgfpHhqvvO-58TUY72rPAWJQ7m4xSiFXD2B8VR-n7ZgleRyvQb3K47Eb-4Tiu70kcPxtEcczxhQTfZB8BpUg3-y_hA'
```
Comment (daimdev) [inline src/test/routes/features/procurement/controllers/procurement.test.ts]: This consist of personal data which is being exposed

**Review by daimdev — CHANGES_REQUESTED** [review]

File: src/test/routes/features/agreement/controllers/selectedAgreement.test.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        .reply(200, { data: {data: {name: 'sisar', ID: mockMenu}} });
+      }
+      nock('https://tst.api.crowncommercial.gov.uk')
+      .post(`/security/tokens/validation`)
+      .query({ 'client-id': 'zyicrDa0oJsH4hULIWTNdadxQV477w45'  })
+      .reply(200, { data: true});
+
+      nock('https://tst.api.crowncommercial.gov.uk')
+      .get(`/user-profiles`)
+      .query({ 'user-id': 'cesar.fernandez@cognizant.com'  })
```
Comment (daimdev) [inline src/test/routes/features/agreement/controllers/selectedAgreement.test.ts]: this bit as well. Please avoid sharing personal details

File: src/test/routes/features/agreement/controllers/agreement.test.ts (hunk 1 above)
Comment (CesarPerez) [reply src/test/routes/features/agreement/controllers/agreement.test.ts]: removed

File: src/test/routes/features/agreement/controllers/selectedAgreement.test.ts (hunk 3 above)
Comment (CesarPerez) [reply src/test/routes/features/agreement/controllers/selectedAgreement.test.ts]: removed

File: src/test/routes/features/procurement/controllers/procurement.test.ts (hunk 4 above)
Comment (CesarPerez) [reply src/test/routes/features/procurement/controllers/procurement.test.ts]: removed

File: src/test/routes/features/agreement/controllers/lot.test.ts (hunk 2 above)
Comment (CesarPerez) [reply src/test/routes/features/agreement/controllers/lot.test.ts]: removed

**Review by daimdev — APPROVED** [review]

File: src/test/routes/features/agreement/controllers/selectedAgreement.test.ts (hunk 5 above)
Comment (CesarPerez) [reply src/test/routes/features/agreement/controllers/selectedAgreement.test.ts]: done

**Pull request merged**